simulations/src/network/mod.rs

```diff
@@ -267,16 +270,31 @@ where
         let to = message.to.expect("adhoc message has recipient");
         if let Some(delay) = self.send_message_cost(rng, message.from, to) {
             let node_capacity = self.node_network_capacity.get(&to).unwrap();
-            if network_time.add(delay) <= self.network_time
-                && node_capacity.increase_load(message.size_bytes)
-            {
+            let should_delay = network_time.add(delay) <= self.network_time;
+            let remaining_size = message.remaining_size();
+            if should_delay && node_capacity.increase_load(remaining_size) {
                 let to_node = self.to_node_senders.get(&to).unwrap();
                 to_node
                     .send(message.clone())
                     .expect("node should have connection");
-                node_capacity.decrease_load(message.size_bytes);
+                node_capacity.decrease_load(remaining_size);
                 return false;
             } else {
+                // if we do not need to delay, then we should check if the msg is too large
+                // if so, we mock the partial sending message behavior
+                if should_delay {
+                    let mut cap = node_capacity.current_load.lock();
+                    let sent = node_capacity.capacity_bps - *cap;
+                    *cap = node_capacity.capacity_bps;
+                    if message.partial_send(sent) == 0 {
+                        let to_node = self.to_node_senders.get(&to).unwrap();
+                        to_node
+                            .send(message.clone())
+                            .expect("node should have connection");
+                        node_capacity.decrease_load(sent);
+                        return false;
+                    }
```

Comment:
Nitpick: can we put this in a named method?

Reply:
Sure!